telegram-history-dump.rb

```diff
@@ -76,7 +76,13 @@ def dump_dialog(dialog)
           msg_chunk = exec_tg_command('history', dialog['print_name'],
                                       $config['chunk_size'], cur_offset)
         end
-        break
+        if msg_chunk.is_a?(Array)
+          break
+        else
+          $log.error('Telegram-cli returned a non array chunk, retrying... (%d/%d)' % [
+            retry_count += 1, $config['chunk_retry']
+          ])
+        end
```

Comment:
Not clear to me: the block of code is inside a `while retry_count <= $config['chunk_retry'] do` so it should never repeat more than `retry_count`.